packages/sol-cov/src/coverage_subprovider.ts

```diff
@@ -115,29 +118,52 @@ export class CoverageSubprovider extends Subprovider {
         const trace: TransactionTrace = jsonRPCResponsePayload.result;
-        const coveredPcs = _.map(trace.structLogs, log => log.pc);
         if (address === constants.NEW_CONTRACT) {
-            const traceInfo: TraceInfoNewContract = {
-                coveredPcs,
-                txHash,
-                address: address as 'NEW_CONTRACT',
-                bytecode: data as string,
-            };
-            this._coverageManager.appendTraceInfo(traceInfo);
+            const tracesByContractAddress = getTracesByContractAddress(trace.structLogs, address);
+            for (const subcallAddress of _.keys(tracesByContractAddress)) {
+                let traceInfo: TraceInfoNewContract | TraceInfoExistingContract;
+                if (subcallAddress === 'NEW_CONTRACT') {
+                    const traceForThatSubcall = tracesByContractAddress[subcallAddress];
+                    const coveredPcs = _.map(traceForThatSubcall, log => log.pc);
+                    traceInfo = {
+                        coveredPcs,
+                        txHash,
+                        address: address as 'NEW_CONTRACT',
+                        bytecode: data as string,
+                    };
+                } else {
+                    payload = { method: 'eth_getCode', params: [subcallAddress, BlockParamLiteral.Latest] };
+                    const runtimeBytecode = (await this.emitPayloadAsync(payload)).result;
+                    const traceForThatSubcall = tracesByContractAddress[subcallAddress];
+                    const coveredPcs = _.map(traceForThatSubcall, log => log.pc);
+                    traceInfo = {
+                        coveredPcs,
+                        txHash,
+                        address: subcallAddress,
+                        runtimeBytecode,
+                    };
+                }
+                this._coverageManager.appendTraceInfo(traceInfo);
+            }
         } else {
-            payload = { method: 'eth_getCode', params: [address, 'latest'] };
-            const runtimeBytecode = (await this.emitPayloadAsync(payload)).result;
-            const traceInfo: TraceInfoExistingContract = {
-                coveredPcs,
-                txHash,
-                address,
-                runtimeBytecode,
-            };
-            this._coverageManager.appendTraceInfo(traceInfo);
+            const tracesByContractAddress = getTracesByContractAddress(trace.structLogs, address);
+            for (const subcallAddress of _.keys(tracesByContractAddress)) {
+                payload = { method: 'eth_getCode', params: [subcallAddress, BlockParamLiteral.Latest] };
+                const runtimeBytecode = (await this.emitPayloadAsync(payload)).result;
+                const traceForThatSubcall = tracesByContractAddress[subcallAddress];
+                const coveredPcs = _.map(traceForThatSubcall, log => log.pc);
+                const traceInfo: TraceInfoExistingContract = {
+                    coveredPcs,
```

Comment:
There is a bunch of duplicate code here with the conditional branch above, can we consolidate?